pkg/common/kubernetes/informers.go

```diff
@@ -30,17 +30,30 @@ func noResyncPeriodFunc() time.Duration {
 	return 0
 }
 
-var signalHandler <-chan struct{}
+var (
+	signalHandler   <-chan struct{}
+	informerFactory informers.SharedInformerFactory
```

Comment:
Not a fan of global var for the informer factory, is there a way to build the informer factory in Initialize() and just pass down only the needed informers down to connection manager (i.e. `informerFactory.Core().Secrets()`)?

Reply:
hmmm okay, I see now that would require a larger refactory on the InformerManager

Reply:
I think it's ok as a global. The global `informerFactory` isn't exported out of the package/process so it can't be re-used by other processes. Since this global is confined to this package, you are getting exactly what you want a single informer factory.

Reply:
If you really do want to pass in the variable, it would require a second function for NewInformer(WithProvidedFactory) and also a second function to InitializeSecretLister(WithProvidedFactory), but it would still accomplish effectively the same thing with a private global.

Reply:
So I think my biggest concern with this is that you would call `informerFactory.Start()` multiple times for multiple `InformerManagers` using the same shared informer factory. Ideally you construct the informer factory in one place and pass that into various types that need resource-specific informers and you would call `informerFactory.Start()` once, ideally in the same place you constructed it. With that said, I think the real problem isn't the global var but the `InformerManager` type which makes it hard to easily pass down the informer factory but probably don't want to address that in this PR.

Reply:
Yea, I looked at the code behind `Start()` and also tested to make sure the behavior is correct. That's why I made the update to the function comment. You can call `Start()` multiple times as you add in new Listers and it only inits the new ones.

The downside of getting all of the Listers upfront, is it complicates the CPI init logic a lot.